src/renderers/shared/fiber/ReactFiberClassComponent.js

```diff
@@ -283,6 +305,7 @@ module.exports = function(
     workInProgress.stateNode = instance;
     // The instance needs access to the fiber so that it can schedule updates
     ReactInstanceMap.set(instance, workInProgress);
+    instance._reactInternalInstance = fakeInternalInstance;
```

Comment:
If it's DEV only we at least know that we can remove it at some point. E.g. a minor.